### DISCUSSION TITLE
docker fails to build 0.4.8 - npm run build error

### DISCUSSION THREAD

**Discussion by NotReallyADeveloper** [open-webui/open-webui#7792] [General]
# Bug Report

## Installation Method

manual docker build . or sh ./run.sh

## Environment

- **Open WebUI Version:0.4.8** 
- **Ollama (if applicable): n/a** 

- **Operating System: Mac OS  15.1.1** 
- **Browser (if applicable):n/a **

**Confirmation:**

- [ x] I have read and followed all the instructions provided in the README.md.
- [ x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ x] I have included the Docker container logs.
- [ x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
docker container build should complete successfully

## Actual Behavior:
process crashes out

## Description

**Bug Summary:**
fatal error when trying to build docker container using 0.4.8 on Mac mini M4 Pro & Sequoia 15.1.1

## Reproduction Details

**Steps to Reproduce:**
./docker build . or sh ./run.sh 

## Logs and Screenshots

```
20.63 <--- JS stacktrace --->
20.63 
20.63 FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
20.63 ----- Native stack trace -----
20.63 
21.17 npm error path /app
21.17 npm error command failed
21.17 npm error signal SIGABRT
21.17 npm error command sh -c npm run pyodide:fetch && vite build
21.17 npm error A complete log of this run can be found in: /root/.npm/_logs/2024-12-11T20_49_24_962Z-debug-0.log
------

 2 warnings found (use docker --debug to expand):
 - SecretsUsedInArgOrEnv: Do not use ARG or ENV instructions for sensitive data (ENV "OPENAI_API_KEY") (line 63)
 - SecretsUsedInArgOrEnv: Do not use ARG or ENV instructions for sensitive data (ENV "WEBUI_SECRET_KEY") (line 63)
Dockerfile:34
--------------------
  32 |     COPY . .
  33 |     ENV APP_BUILD_HASH=${BUILD_HASH}
  34 | >>> RUN npm run build
  35 |     
  36 |     ######## WebUI backend ########
--------------------
ERROR: failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/rm97wq99p6xjdj74hknmbbo8x
Unable to find image 'open-webui:latest' locally
docker: Error response from daemon: pull access denied for open-webui, repository does not exist or may require 'docker login'.
See 'docker run --help'.
Total reclaimed space: 0B

```

**Comment by NotReallyADeveloper** [comment]
As my user name suggests, I am not a developer.  But I did manage to solve this.

Insert before line 34: 

`ENV NODE_OPTIONS=--max-old-space-size=4096`

Then make sure docker has at least 8 GB memory available to it.

docker build . ran perfectly to build the image.

To run the image, I ran the following command (modified from what is in the README file)

`docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 --name open-webui --restart always open-webui:latest`

Hope this helps (don't know how to submit to the codebase, sorry).

**Comment by tjbck** [comment]
Please check for previous discussions before raising one.

**Reply by NotReallyADeveloper** [reply]
I did but didn't see this anywhere.  Sorry.

**Comment by carlosnatalino** [comment]
For those coming here due to the same issue (like me), this is the direct link to the documentation of this issue: https://docs.openwebui.com/getting-started/advanced-topics/development#fatal-error-reached-heap-limit
